shell/platform/tizen/touch_event_handler.cc

```diff
@@ -74,38 +74,51 @@ Eina_Bool TouchEventHandler::OnTouch(void* data, int type, void* event) {
   auto* self = reinterpret_cast<TouchEventHandler*>(data);
 
   if (type == ECORE_EVENT_MOUSE_BUTTON_DOWN) {
-    self->pointer_state_ = true;
     auto* button_event = reinterpret_cast<Ecore_Event_Mouse_Button*>(event);
-    self->SendFlutterPointerEvent(kDown, button_event->x, button_event->y, 0, 0,
-                                  button_event->timestamp,
-                                  button_event->multi.device);
+    if (self->engine_->renderer()->GetWindowId() == button_event->window) {
+      self->pointer_state_ = true;
+      self->SendFlutterPointerEvent(kDown, button_event->x, button_event->y, 0,
+                                    0, button_event->timestamp,
+                                    button_event->multi.device);
+      return ECORE_CALLBACK_DONE;
+    }
+
   } else if (type == ECORE_EVENT_MOUSE_BUTTON_UP) {
-    self->pointer_state_ = false;
     auto* button_event = reinterpret_cast<Ecore_Event_Mouse_Button*>(event);
-    self->SendFlutterPointerEvent(kUp, button_event->x, button_event->y, 0, 0,
-                                  button_event->timestamp,
-                                  button_event->multi.device);
+    if (self->engine_->renderer()->GetWindowId() == button_event->window) {
```

Comment:
The id may be stored to a variable and reused.

```cpp
auto window_id = self->engine_->renderer()->GetWindowId();
```

Reply:
You're right! that would be better